### PR TITLE
feat: set county on building address

### PR DESCRIPTION
**Pull request by chriscasto**
# Pull Request Template

## Issue Overview

This PR addresses #102

- [X] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

This PR modifies the transformer function for `buildingAddress` to assign a county to the address does not already exist according to the business logic laid out in the [Doorway jurisdiction design doc](https://docs.google.com/document/d/1kyVX9cJUTa3XcUBWKkudeE8X2BgpvCZs6ww07NaM4bc/edit#heading=h.5ku3nid3w6da).

## How Can This Be Tested/Reviewed?

Unit tests have been added, so just run `yarn test`.

## Checklist:

- [X] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [X] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [X] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: tasks/import-listings/src/etl/transform/map.ts [hunk 1]
```diff
@@ -52,7 +52,36 @@ export const defaultMap: RecordMap = {
...
+
+    // if we don't have an address, ignore
+    if (!address) return
+
+    if (!address?.county) {
+      const jurisdiction = listing.jurisdiction
+
+      if (jurisdiction.name == "San Jose") {
+        address.county = "Santa Clara"
+      }
```
Comment (phattya) [inline tasks/import-listings/src/etl/transform/map.ts:66]: Why is this needed here, if it's also part of the switch statement below? Seems repetitive, but maybe I'm missing something.

**Review by phattya — APPROVED** [review]
Looks good to me. Added a few non-blocking comments.

File: tasks/import-listings/src/etl/transform/map.ts (hunk 1 above)
Comment (chriscasto) [reply tasks/import-listings/src/etl/transform/map.ts:66]: You're right! I initially added the conditions as individual conditions just in case the test logic differed, but I ended up being able to put them all into a switch instead.  I'll remove it.

File: tasks/import-listings/test/etl/transformer.spec.ts [hunk 2]
```diff
@@ -209,4 +209,52 @@ describe("Transformer", () => {
...
+      },
+      {
+        // specific to San Mateo
+        jurisdiction: "San Mateo",
+        expect: "San Mateo",
+      },
+      {
+        // default behavior
+        jurisdiction: "Detroit",
+        expect: "Detroit",
```
[the remark this answers is not in the document]
Comment (chriscasto) [reply tasks/import-listings/test/etl/transformer.spec.ts:236]: This is just a test to make sure that the default condition in the switch applies correctly.

**Comment by chriscasto** [comment]
@lisunshiny I'm not sure why the linter or unit tests would be failing since I didn't touch anything outside of the task folder.  I ran `yarn lint` in the root and it didn't have any errors.  I ran `yarn test:unit` on `sites/partners` and it came back clean too.

Would you mind taking a look?

**Comment by chriscasto** [comment]
@lisunshiny Disregard previous request.  The errors disappeared when the checks re-ran.  Must have been something else.

**Review by ifranch — APPROVED** [review]

**Review by lisunshiny — APPROVED** [review]
